python/lsst/daf/butler/datastores/fileDatastore.py

```diff
@@ -1648,59 +1607,43 @@ def emptyTrash(self, ignore_errors: bool = True) -> None:
         """
         log.debug("Emptying trash in datastore %s", self.name)
         # Context manager will empty trash iff we finish it without raising.
-        with self.bridge.emptyTrash() as trashed:
-            for ref in trashed:
-                fileLocations = self._get_dataset_locations_info(ref)
-
-                if not fileLocations:
-                    err_msg = f"Requested dataset ({ref}) does not exist in datastore {self.name}"
-                    if ignore_errors:
-                        log.warning(err_msg)
-                        continue
-                    else:
-                        raise FileNotFoundError(err_msg)
-
-                for location, _ in fileLocations:
-
-                    if not self._artifact_exists(location):
-                        err_msg = f"Dataset {location.uri} no longer present in datastore {self.name}"
+        # It will also automatically delete the relevant rows from the
+        # trash table and the records table.
+        with self.bridge.emptyTrash(self._table, record_class=StoredFileInfo) as trashed:
```

Comment:
@TallJimbo this is the first pass at doing the join in the bridge. It still does the per-file test though. For this implementation I could assume the list of records is manageable and do an `IN` select on all the files to get back the refs in a set and compare the two. I didn't quite catch your idea in the meeting.